src/tdastro/astro_utils/passbands.py

```diff
@@ -18,36 +17,55 @@ class PassbandGroup:
 
     Attributes
     ----------
-    passbands : dict
-        A dictionary of Passband objects.
+    passbands : dict of Passband
+        A dictionary of Passband objects, where the keys are the full_names of the passbands (eg, "LSST_u").
+    waves : np.ndarray
+        The union of all wavelengths in the passbands.
     """
 
-    def __init__(self, preset=None, passbands=None):
-        """Initialize a PassbandGroup object.
-
-        Attributes
-        ----------
-        preset : str, optional
-            A pre-defined set of passbands to load.
-        passbands : dict
-            A dictionary of Passband objects assigned to the group. The keys are the passbands' full names:
-            eg, LSST_u, LSST_g, HSC_u, etc.
+    def __init__(
+        self,
+        preset: str = None,
+        passband_parameters: list = None,
+        **kwargs,
+    ):
+        """Construct a PassbandGroup object.
 
         Parameters
         ----------
         preset : str, optional
             A pre-defined set of passbands to load.
-        passbands : list, optional
-            A list of Passband objects assigned to the group.
+        passband_parameters : list of dict, optional
+            A list of dictionaries of passband parameters used to create Passband objects.
+            Each dictionary must contain the following:
+            - survey : str
+            - filter_name : str
+            Dictionaries may also contain the following optional parameters:
+            - table_path : str
+            - table_url : str
+            - delta_wave : float
+            - trim_quantile : float
+            - units : str (either 'nm' or 'A')
+            If survey is not LSST (or other survey with defined defaults), either a table_path or table_url
+            must be provided.
+        **kwargs
+            Additional keyword arguments to pass to the Passband constructor.
         """
         self.passbands = {}
 
         if preset is not None:
-            self._load_preset(preset)
+            self._load_preset(preset, **kwargs)
+
+        if passband_parameters is not None:
```

Comment:
Do you want to raise an error if this is None? Or does the class use reasonable defaults in that case?